core/src/main/java/org/apache/iceberg/rest/RestCatalog.java

```diff
@@ -0,0 +1,324 @@
+
+    this.mapper = new ObjectMapper();
+    RequestResponseSerializers.registerAll(mapper);
+
+    // Pass in the mock client instead.
+    this.restClient = httpClient;
+  }
+
+  @Override
+  public String name() {
+    return catalogName;
+  }
+
+  @Override
+  public Table createTable(
+      TableIdentifier identifier, Schema schema, PartitionSpec spec, String location,
+      Map<String, String> props) {
+    throw new UnsupportedOperationException("Not implemented: createTable");
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    // Users might have not authenticated, possibly need to reauthenticate, or get authentication info per request.
+    //   Though usually, auth tokens are persistent for at least some period of time.
+    return newTableOps(tableIdentifier, null);
+  }
+
+  protected RestTableOperations newTableOps(TableIdentifier tableIdentifier, String authToken) {
+    throw new UnsupportedOperationException("Not implemented: newTableOps");
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+    throw new UnsupportedOperationException("Not implemented: defaultWarehouseLocation");
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    throw new UnsupportedOperationException("Not implemented: listTables");
+  }
+
+  @Override
+  public boolean tableExists(TableIdentifier identifier) {
+    throw new UnsupportedOperationException("Not implemented: tableExists");
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier) {
+    return dropTable(identifier, false);
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    throw new UnsupportedOperationException("Not implemented: dropTable");
+  }
+
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    throw new UnsupportedOperationException("Not implemented: renameTable");
+  }
+
+  @Override
+  public void close() throws IOException {
+    restClient.close();
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    CreateNamespaceRequest req = CreateNamespaceRequest.builder()
+        .withNamespace(namespace)
+        .withProperties(metadata)
+        .build();
+
+    // TODO - This should come from the server side.
+    String path = properties.getOrDefault("create-namespace-path", "namespace");
+    restClient.post(path, req, CreateNamespaceResponse.class, ErrorHandlers.namespaceErrorHandler());
+  }
+
+  public CreateNamespaceResponse createDatabase(Namespace namespace, Map<String, String> props) {
+    CreateNamespaceRequest req = CreateNamespaceRequest.builder()
+        .withNamespace(namespace)
+        .withProperties(props)
+        .build();
+
+    // TODO - This should come from the server side.
+    String path = properties.getOrDefault("create-namespace-path", "namespace");
```

Comment:
what is this about?

Reply:
This is left over from a previous idea I was toying with.

I will remove this and others like it. It would be pretty hard to deal with it from the iceberg library if users were rewriting the path, given that it contains path variables and things at times. This is just an artifact I forgot to remove. Good catch.